calculator_test.go

```diff
@@ -2,23 +2,304 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
 func TestAdd(t *testing.T) {
-	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+
+	t.Run("Fixed number addition", func(t *testing.T) {
+		var want float64 = 4
+		got := calculator.Add(2, 2)
+		if want != got {
+			t.Errorf("want %f, got %f", want, got)
+		}
+	})
+
+	t.Run("Sum four numbers", func(t *testing.T) {
+		var want float64 = 40
+		got := calculator.Add(2, 8, 20, 10)
+		if want != got {
+			t.Errorf("want %f, got %f", want, got)
+		}
+	})
+
+}
+
+func TestAddRandom(t *testing.T) {
+	var totalRandom int = 100
+	t.Run("Random number addition", func(t *testing.T) {
+		for count := 0; count < totalRandom; count++ {
+			var a float64 = rand.NormFloat64()*1024 + 2
+			var b float64 = rand.NormFloat64()*2048 + 5
+			want := a + b
+			got := calculator.Add(a, b)
+			if want != got {
+				t.Errorf("want %f, got %f", want, got)
+			}
+		}
+	})
+}
+
+func TestSubtract(t *testing.T) {
+	testCases := []struct {
+		name  string
+		want  float64
+		n1    float64
+		n2    float64
+		extra []float64
+	}{
+		{
+			name:  "Substract a > b returns positive",
```

Comment:
```suggestion
			name:  "Subtract a > b returns positive",
```

Always worth using a spell checker! (The SpellRight extension for VS Code is good.)